src/components/table/table.tsx

```diff
@@ -205,6 +213,37 @@ export class Table {
         this.changePage.emit(page);
     }
 
+    private onClickRow(event: MouseEvent, row: Tabulator.RowComponent): void {
+        // deselect all activated rows
+        this.tabulator.deselectRow();
+        row.select();
+        const object = row.getData();
+
+        const activatedObject: any = this.data.find((item: any) => {
+            return !!item.activated;
```

Comment:
We can not do it like this. This forces the consumer to have a field in their data that is named `activated`. If we want to store the `activated` state with the data, we need an interface for the data and also nest the actual data deeper in that structure since the consumer might have a field named `activated` that is used for something else. Also, what happens if the consumer sets `activated` to true on all the rows? As a consumer I would expect all rows to be activated, but nothing will happen. Even if I just set one of the rows to active when data is given to the table, it will not activate the row since we only have logic for it when we are clicking.

How about adding a new `@Prop` to the table that tells us what row is currently active? That way we can have the user activate a row by changing the input, and the event will tell the user when a new row has been activated.